nexus/src/app/background/init.rs

```diff
@@ -418,13 +421,32 @@ impl BackgroundTasksInitializer {
         let blueprint_loader =
             blueprint_load::TargetBlueprintLoader::new(datastore.clone());
         let rx_blueprint = blueprint_loader.watcher();
+
+        // Background task: blueprint planner
+        let blueprint_planner = blueprint_planner::BlueprintPlanner::new(
+            datastore.clone(),
+            false,
+            rx_blueprint.clone(),
+        );
+        let rx_planner = blueprint_planner.watcher();
+        driver.register(TaskDefinition {
+            name: "blueprint_planner",
+            description: "Updates the target blueprint",
+            period: config.blueprints.period_secs_plan,
+            task_impl: Box::new(blueprint_planner),
+            opctx: opctx.child(BTreeMap::new()),
+            watchers: vec![Box::new(rx_blueprint.clone())],
```

Comment:
How about adding the inventory collector here?  That way we'll re-evaluate the blueprint if the inventory (may have) changed.  I'm on the fence about whether we should also use the latest inventory from the `inventory_collection` task instead of fetching it.  If we keep this as is, then we could always revisit, maybe under #5296.

Reply:
Thanks, updated in 33d02e4fd. Since we're using the `inventory_collection` task as a notifier, I think it makes sense to use the collection it reports. This also matches how we treat the current target blueprint as whatever the `blueprint_loader` task reports.